exporter/awskinesisexporter/internal/batch/encode_marshaler.go

```diff
@@ -61,33 +62,58 @@ func (bm *batchMarshaller) Logs(ld plog.Logs) (*Batch, error) {
 func (bm *batchMarshaller) Traces(td ptrace.Traces) (*Batch, error) {
 	bt := New(bm.batchOptions...)
 
+	var errs error
+
+	if bm.partitioner.Keyed() {
```

Comment:
Would it be possible to rework this so we don't have to effectively have the same code twice and make it easier to maintain?

Reply:
Refactored to simplify Trace handling for both keyed and non-keyed partitioning in single loop and avoid code duplication.